zerver/lib/event_queue.py

```diff
@@ -125,6 +133,7 @@ def add_event(self, event):
         self.finish_current_handler()
 
     def finish_current_handler(self, need_timeout=False):
+        # type: (Optional[bool]) -> bool
```

Comment:
Shouldn't `need_timeout` be annotated as `bool`? I don't think `None` should be a valid value for `need_timeout`.

You might be mistaken about the meaning of `Optional`. `Optional[T]` is supposed to be the same as `Union[None, T]`.